### PR TITLE
chore(doc): move command docs from legacy /doc to cmd/doc.go

### PR DESCRIPTION
**Pull request by nvinuesa**
This small patch removes the legacy doc/commands.txt in favor of a new cmd/doc.go standard package document. Also removes the execution section.

## Checklist

- [ ] ~Code style: imports ordered, good names, simple structure, etc~
- [ ] ~Comments saying why design decisions were made~
- [ ] ~Go unit tests, with comments saying what you're testing~
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [X] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps
N/A

## Documentation changes

New `cmd/doc.go` package doc.

## Links

**Jira card:** JUJU-4992

### CODE REVIEW

File: cmd/doc.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Commands and Sub-commands
+// =========================
+// Commands need to provide an `Info` method that returns an Info struct.
+
```
Comment (SimonRichardson) [inline cmd/doc.go:7]: This should start with `Package cmd...`

Also remove the weird markdown formatting.

Comment (tmihoc) [reply cmd/doc.go:7]: +1. Plus I'm guessing that means removing the markdown formatting throughout the remainder of the doc.

File: cmd/doc.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// `Supercommand`s are commands that do many things, and have "sub-commands" that
+// provide this functionality.  Git and Bazaar are common examples of
+// "supercommands".  Subcommands must also provide the `Command` interface, and
+// are registered using the `Register` method.  The name and aliases are
+// registered with the supercommand.  If there is a duplicate name registered,
+// the whole thing panics.
+
+// Supercommands need to be created with the `NewSuperCommand` function in order
+// to provide a fully constructed object.
+
```
Comment (tmihoc) [inline cmd/doc.go:27]: suggestion: Add an example of a supercommand. Also mention that they are rare in `juju` -- we mostly don't use that pattern.

File: cmd/doc.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// The 'help' subcommand
+// ---------------------
+// All supercommand instances get a help command.  This provides the basic help
+// functionality to get all the registered commands, with the addition of also
+// being able to provide non-command help topics which can be added.
+
+// Help topics have a `name` which is what is matched from the command line, a
+// `short` one line description that is shown when `<cmd> help` is called,
+// and a `long` text that is output when the topic is requested.
+package cmd
```
Comment (tmihoc) [inline cmd/doc.go:37]: This was confusing to me until I figured out it's describing the 'help' on supercommands, e.g., `juju wait-for help application`. 

This bit still doesn't make any sense to me:

> This provides the basic help
// functionality to get all the registered commands, with the addition of also
// being able to provide non-command help topics which can be added.

suggestion: Remove + rephrase the first sentence:

> // Every supercommand gets a help command which provides the basic help functionality to 
> // its respective registered subcommands.

**Review by tmihoc — CHANGES_REQUESTED** [review]
For Juju 3.4: Merge the supercommand and help sections, as the help is describing something about supercommands only.

For Juju 4: We should check if there is any supercommand left. If not, we should put up another PR to remove both the supecommand and the help section.

**Review by tmihoc — APPROVED** [review]
If you make the changes we discussed (the example and the rephrase), LGTM.

File: cmd/doc.go (hunk 2 above)
Comment (nvinuesa) [reply cmd/doc.go:28]: fixed

File: cmd/doc.go (hunk 3 above)
Comment (nvinuesa) [reply cmd/doc.go:36]: done, also rebased and changed target branch to main

**Pull request merged**